src/DocumentDB.ChangeFeedProcessor/ChangeFeedProcessorBuilder.cs

```diff
@@ -464,5 +512,34 @@ private async Task InitializeCollectionPropertiesForBuildAsync()
             this.databaseResourceId = this.databaseResourceId ?? await GetDatabaseResourceIdAsync(this.feedDocumentClient, this.feedCollectionLocation).ConfigureAwait(false);
             this.collectionResourceId = this.collectionResourceId ?? await GetCollectionResourceIdAsync(this.feedDocumentClient, this.feedCollectionLocation).ConfigureAwait(false);
         }
+
+        /// <summary>
+        /// Internal metadata/attributes of the collection object.
+        /// </summary>
+        private class CollectionMetadata
```

Comment:
nit: Shouldn't it have it's own file?

Reply:
should be fine for inner classes. Another way would be to have parent class as partial and this one in separate file, inside partial class, seems to be too much.